skimage/util/dtype.py

```diff
@@ -321,6 +321,15 @@ def _convert(image, dtype, force_copy=False, uniform=False):
             # DirectX uses this conversion also for signed ints
             # if imin_in:
             #     np.maximum(image, -1.0, out=image)
+        elif kind_in == 'i':
```

Comment:
@scikit-image/core Can anyone verify that this part of the patch is correct?

Reply:
@stefanv What you're  actually concern about in this part? Maybe I should give you some details, so you would be able to answer by yourself?

Reply:
Yes, this is some of the most finicky code in the library, so it is helpful to have a very good sense of what is happening.

Reply:
First of all: As I see kind_in == 'i' stand for all signed integer types.
Before that signed integer types went all to else block where they get first added 0.5 before normalizing to range [-1.0 ; 1.0].
That for first look seems to be good trick to get off problems of their ranges since e.g int8 has [-128 ; 127] and imax_out will assume as its value 127. 
But in that way all normalized signed integers cannot achieve exact 0 value, which for future calculations will lead to lots of problems and unnecessary lost features. Also note as adding 0.5 will affect more normalization as number of bytes of base int will drop and for int8 it especially hits hard.
I've performed conversion from DirectX documentation. It is still a bit leaky cause -2^bites and -(2^bites - 1) will both be normalized to -1.0, but that won't affect as much and rarely will be noticable

Reply:
Imagine for example that you got single x = 0 with type int8.
with else block you will first have x = 0.5 since addition
then it will be multiplied to x = 2 and divided to get x = 2/255.
You see that something is wrong, but when you have e.g int64 you'll get 2/9223372036854775807 which is small enough to fool you.

With added elif you get for int8 x = 0/127 and for int64 x = 0/9223372036854775807 which is still x=0.

Reply:
Thanks @michalkrawczyk, this change does indeed look consistent with the Direct X document you linked to and seems reasonable to me.